src/test/crimson/test_config.cc

```diff
@@ -1,26 +1,75 @@
 #include <chrono>
+#include <string>
 #include <numeric>
 #include <seastar/core/app-template.hh>
 #include <seastar/core/sharded.hh>
+#include "common/config_obs.h"
 #include "crimson/common/config_proxy.h"
 
 using Config = ceph::common::ConfigProxy;
+const std::string test_uint_option = "osd_max_pgls";
+
+class ConfigObs : public ceph::internal::md_config_obs_impl<Config> {
+  uint64_t last_change;
+  uint64_t num_changes = 0;
+  std::unique_ptr<const char*[]> conf_keys = std::make_unique<const char*[]>(2);
```

Comment:
could really just be `const char* conf_keys[] = { nullptr, nullptr };`

Reply:
Hmm.. This cannot be compiled. I'm a little confused by the const pointers here. I think this is because `get_tracked_conf_keys()` has a `const` qualifier, so `conf_keys` pointers cannot be stored directly in `ConfigObs`.
```
error: invalid conversion from 'const char* const*' to 'const char**' [-fpermissive]
     return conf_keys;
```

Reply:
agreed! the `const` specifier marks the returned member variable of `get_tracked_conf_keys()` `const`. but another solution could be something like
```c++
  const char** get_tracked_conf_keys() const override {
    static const char* keys[] = {
      test_uint_option.c_str(),
      nullptr,
    };
    return keys;
  }
```

because `test_uint_option` is a global variable and is initialized before we call this method for the first time. so we can safely initialize `keys` this way.